Sources/SymbolKit/SymbolGraph/Relationship/Relationship.swift

```diff
@@ -106,3 +106,30 @@ extension SymbolGraph {
         }
     }
 }
+
+extension SymbolGraph.Relationship: Hashable, Equatable {
```

Comment:
Is there a relationship property that's not hashable which prevents the compiler from synthesizing hashable and equatable implementations?

Ideally we'd use the synthesized implementations so that these implementations don't go out of date when new properties are added.

Reply:
I _think_ this has to do with the way that Symbols and Relationships have the `[String: Mixin]` map - the Mixin protocol doesn't require Hashable/Equatable, so they have to be pulled out by hand.

Reply:
> the Mixin protocol doesn't require Hashable/Equatable

Should it? It would be a source breaking change to add—since non-hashable types wouldn't conform anymore—it but maybe it's worth it.

Reply:
I think it would make sense. We already implement these for mixins defined within SymbolKit (hence the implementation here), so adding the requirement would make sense there.

Reply:
It turns out that adding a `Hashable` requirement to `Mixin` causes SymbolKit to fail to build; it makes it so that it can only be used as a type name on Swift 5.7, as `any Mixin`. Since we support Swift 5.2 for SymbolKit (and Swift 5.7 isn't officially released yet) this would be a massively breaking change.